grcs/0020-knowledge-graph.md

```diff
@@ -32,164 +32,176 @@ Data is encoded using Protobufs. Protocol Buffers are a language-neutral, platfo
 
-Additional attributes can be added to Number attributes to specify hints for database optimizations. This can allow application developers to add additional constraints that can be useful for choosing more efficient storage types.
+| Name         | ID                     | Description                                               |
+| -------------| ---------------------- | --------------------------------------------------------- |
+| Text         | 992321fd-c288-4dac-bf89-27f52e448557 | A string of characters                                    |
+| Number       | eb3155c5-7e70-406f-ae70-17f029a6a76d | A decimal value                                           |
+| Checkbox     | c6345346-2e40-456f-a7d8-ffdbdd81fec7 | Can either be `true` or `false`                           |
+| URL          | bb8c1304-7f3a-43e7-a444-d3aa589fb096 | A URI to a resource with a protocol identifier            |
+| Time         | 3c62afbd-c430-438a-83b0-61c60de5f03f | A date and time or interval                               |
+| Point        | bc76a151-fa46-4a9a-9590-c40e1738f452 | For geo locations or other coordinates                    |
 
-#### 5.3 Checkbox
 
-A checkbox can either be `true` or `false`. It's a boolean field named for normal people. The string encoding of the checkbox field must be "1" for `true` and "0" for `false`. Checkbox values can be rendered with alternate UI components such as switches and toggles.
+In addition, properties can have Value Type: Relation, which isn't a native type, but the allows us to define relation types as part of the schema. For each native value type, if a value is provided that is invalid, the triple should be dropped and treated as unset. Null cannot be provided as a value. Each native value type is UTF-8 encoded as a string.
 
-#### 5.4 URL
+### 8. Relations
 
-A URL value type is technically a URI that starts with the protocol identifier followed by a :// and the resource. The initial supported protocols are: graph, ipfs, ar, https.
+Relations describe the edges of a graph. For example a Company can have Team Members. Each Team Member relation can have a property describing when the person joined the team. This is a model that is commonly called a property graph. Relations are enocded in 2 parts: a lightweight relation and an optional relation entity with details about the relationship. 
 
-Examples:
-* "ipfs://QmVGF2e9WqF8g39W81eCQxecz7Bdi5o2DFSJ5BQxwfveYB"
-* "ht<span>tps://</span>geobrowser.io"
+Relations are ordered. They can be reordered independently without conflicts using Fractional Indexing with an arbitrary level of precision between the indices of the items the user wants to move the item between. A fractional index must include randomness to allow items to be inserted between items with minimal chance of collision. This way moving the order of an item doesn't require changing the index of any of the other items.
```

Comment:
Byron: Should we define a specific fractional indexing spec here? Or should it be in some different GRC?